packages/synapse-react-client/src/components/ChangePassword/useChangePasswordFormState.tsx

```diff
@@ -98,7 +98,30 @@ export default function useChangePasswordFormState(
     },
     [changePassword],
   )
+  // Used for client-side validation.  Note, the server has the final say
+  // See https://github.com/Sage-Bionetworks/Synapse-Repository-Services/blob/develop/services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/password/PasswordValidatorImpl.java#L47 for rules
+  const validatePassword = useCallback((newPassword: string) => {
+    if (newPassword.trim().length < 8) {
+      return 'A valid password must be at least 8 characters long'
+    }
+    const hasLetter = /[a-zA-Z]/.test(newPassword) // Checks for at least one letter
+    if (!hasLetter) {
+      return 'A valid password must include letters'
+    }
 
+    const hasNumber = /\d/.test(newPassword) // Checks for at least one number
+    if (!hasNumber) {
+      return 'A valid password must include digits (0-9)'
+    }
+
+    const hasSpecialChar = /[~!@#$%^&*_\-+=`|\\(){}[\]:;"'<>,.?/]/.test(
+      newPassword,
+    ) // Checks for at least one special character
+    if (!hasSpecialChar) {
+      return 'A valid password must include special characters ~!@#$%^&*_-+=`|\\(){}[]:;"\'<>,.?/'
+    }
+    return undefined
+  }, [])
```

Comment:
This function should be pulled out of the hook (ok to keep it in this file), since it doesn't rely on any hook state/effects. It could be unit tested without instantiating the entire hook (but I think the tests you have are comprehensive enough for this new functionality, and I don't think we have unit tests for this hook to begin with).

The function doesn't even need to be returned by the hook, but I recognize that there is value in co-locating it with the stateful password utilities, so that decision makes sense.